Lib/fontTools/subset/svg.py

```diff
@@ -0,0 +1,235 @@
+                # ignore dangling reference; not our job to validate svg
+                continue
+            referenced.update(iter_referenced_ids(elements[el_id]))
+        referenced -= element_ids
+        element_ids.update(referenced)
+        unvisited = referenced
+
+
+def subset_elements(el: etree.Element, ids: Set[str]) -> bool:
+    # Keep elements if their id is in the subset, or any of their children's id is.
+    # Drop elements whose id is not in the subset, and either have no children,
+    # or all their children are being dropped.
+    el_id = el.attrib.get("id")
+    if el_id is not None and el_id in ids:
+        # if id is in the set, don't recurse; keep whole subtree
+        return True
+    keep = False
+    for e in el:
+        keep |= subset_elements(e, ids)
+    if keep:
+        return True
+    assert len(el) == 0
+    parent = el.getparent()
+    if parent is not None:
+        parent.remove(el)
+    return False
+
+
+def remap_glyph_ids(
+    elements: Dict[str, etree.Element], glyph_index_map: Dict[int, int]
+) -> Dict[str, str]:
+    # Given {old_gid: new_gid} map, rename all elements containing id="glyph{gid}"
+    # special attributes
+    id_map = {}
+    for el_id, el in elements.items():
+        m = GID_RE.match(el_id)
+        if not m:
+            continue
+        old_index = int(m.group(1))
+        new_index = glyph_index_map.get(old_index)
+        if new_index is not None:
+            if old_index == new_index:
+                continue
+            new_id = f"glyph{new_index}"
+        else:
+            # If the old index is missing, the element correspond to a glyph that was
+            # excluded from the font's subset.
+            # We rename it to avoid clashes with the new GIDs or other element ids.
+            new_id = f".{el_id}"
+            n = count(1)
+            while new_id in elements:
+                new_id = f"{new_id}.{next(n)}"
+
+        id_map[el_id] = new_id
+        el.attrib["id"] = new_id
+
+    return id_map
+
+
+def update_glyph_href_links(svg: etree.Element, id_map: Dict[str, str]) -> None:
+    # update all xlink:href="#glyph..." attributes to point to the new glyph ids
+    for el in find_svg_elements_with_glyph_href(svg):
+        old_id = el.attrib[XLINK_HREF][1:]
+        if old_id in id_map:
+            new_id = id_map[old_id]
+            el.attrib[XLINK_HREF] = f"#{new_id}"
+
+
+def ranges(ints: Iterable[int]) -> Iterator[Tuple[int, int]]:
+    # Yield (min, max) ranges of consecutive integers from the input set
+    sorted_ints = sorted(set(ints))
+    # to group together consecutive ints, we use as 'key' the difference
+    # between their index in the (sorted) list and themselves, which stays
+    # the same for consecutive numbers
+    for _key, group in groupby(enumerate(sorted_ints), lambda i: i[0] - i[1]):
+        consecutive_ints = [v for _i, v in group]
+        yield (consecutive_ints[0], consecutive_ints[-1])
```

Comment:
After sorting my instinct is to think you could use a reduce rather than a groupby. You add the new value to the last list in the accumulator if the last list is empty or it's last value is v-1 type of deal.

Reply:
Nothing, my thought would just be to use the simplest tool that can solve the problem. Contrasting the "roughly equivalent to" for reduce vs group by would leave me thinking of reduce as "lighter" than groupby.

Reply:
In the end I changed it to use neither groupby nor reduce, just a simple generator that yields (start, end) when v-1 != end https://github.com/fonttools/fonttools/pull/2452/commits/55b2d4bc3b8228814533fc9a2cde7811353dc302